app/api/register/route.ts

```diff
@@ -1,15 +1,39 @@
 import { NextResponse } from "next/server";
+import { createClient } from '@supabase/supabase-js';
 // for sending an email verification message
 // import crypto from 'crypto';
 // import { sendVerificationEmail } from '@/app/lib/sendEmail';
 
+// Start the supabase client
+const supabase = createClient( process.env.SUPABASE_URL, process.env.SUPABASE_ANON_KEY);
+
+// Function returns if inserting the data was successfull, otherwise includes the error message
+export async function registerUser( email: string, password: string): Promise<{ success: boolean; error?: string }> {
+    const { data, error } = await supabase
+        .from('users')
+        .insert([
+            {
+                email: email,
+                password: password,
+                verified: false,
```

Comment:
```suggestion
                verification: 0,
```
the verification status an integer of range [0-2]:
0 = unverified
1 = verified
2 = admin